homeassistant/components/http/__init__.py

```diff
@@ -216,29 +218,36 @@ async def async_setup(hass, config):
         ssl_profile=ssl_profile,
     )
 
-    async def stop_server(event):
+    startup_listeners = []
+
+    async def stop_server(event: Event) -> None:
         """Stop the server."""
         await server.stop()
 
-    async def start_server(event):
+    async def start_server(event: Event) -> None:
         """Start the server."""
+
+        for listener in startup_listeners:
+            listener()
+
         hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_server)
-        await server.start()
 
-        # If we are set up successful, we store the HTTP settings for safe mode.
-        store = storage.Store(hass, STORAGE_VERSION, STORAGE_KEY)
+        await start_http_server_and_save_config(hass, dict(conf), server)
 
-        if CONF_TRUSTED_PROXIES in conf:
-            conf_to_save = dict(conf)
-            conf_to_save[CONF_TRUSTED_PROXIES] = [
-                str(ip.network_address) for ip in conf_to_save[CONF_TRUSTED_PROXIES]
-            ]
-        else:
-            conf_to_save = conf
+    async def async_wait_frontend_load(event: Event) -> None:
+        """Wait for the frontend to load."""
 
-        await store.async_save(conf_to_save)
+        if event.data[ATTR_COMPONENT] != "frontend":
+            return
+
+        await start_server(event)
```

Comment:
Can we end up calling this twice?

Reply:
As in, once via this event handler and once from the start event ?

Reply:
Both listeners get canceled before this is called
https://github.com/home-assistant/core/pull/36093/files#diff-829fc3a9a7a79507cc486a65c18116eeR230-R231